src/main/java/com/coherentsolutions/pot/insuranceservice/controller/AdminEnrollmentManagementController.java

```diff
@@ -15,20 +19,26 @@
 
 @RestController
 @RequiredArgsConstructor
-@RequestMapping("/v1/enrollments")
+@RequestMapping("/v1/companies/{companyId}/users/{userId}/enrollments")
 public class AdminEnrollmentManagementController {
 
   private final EnrollmentManagementService enrollmentService;
 
+  @PreAuthorize("@companyAdminSecurityService.canAccessCompanyResource(#companyId, 'ROLE_FUNC_COMPANY_ENROLLMENT_MANAGER')")
   @PostMapping
   @ResponseStatus(HttpStatus.CREATED)
-  public EnrollmentDto createEnrollment(@Valid @RequestBody EnrollmentDto dto) {
+  public EnrollmentDto createEnrollment(@PathVariable UUID companyId, @PathVariable UUID userId, @Valid @RequestBody EnrollmentDto dto) {
     return enrollmentService.createEnrollment(dto);
   }
 
+  @PreAuthorize(
+      "@userSecurityService.canAccessUserResource(#userId, 'ROLE_FUNC_CONSUMER') "
+          + "or "
+          + "@companyAdminSecurityService.canAccessCompanyResource(#companyId, 'ROLE_FUNC_COMPANY_ENROLLMENT_MANAGER')"
+  )
   @GetMapping
```

Comment:
```suggestion
  @PreAuthorize(
      "@userSecurityService.canAccessUserResource(#companyId, 'ROLE_FUNC_CONSUMER') "
          + "or "
          + "@companyAdminSecurityService.canAccessCompanyResource(#companyId, 'ROLE_FUNC_COMPANY_ENROLLMENT_MANAGER')"
  )
  @GetMapping
```
Then, in the `EnrollmentService#getAll`, you need to check if the current user has a consumer role and add filtering by the logged-in user ID in this case.